drivers/SmartThings/zigbee-fan/src/itm-fan-light/init.lua

```diff
@@ -0,0 +1,182 @@
+  local dev = device:get_parent_device()
+  if dev == nil then
+    dev = device
+  end
+  dev:send(FanControl.attributes.FanMode:read(dev))
+  dev:refresh(driver, dev)
+end
+
+-- CAPABILITY HANDLERS
+
+local function on_handler(driver, device, command)
+  local dev = device
+  if device.network_type == stDevice.NETWORK_TYPE_CHILD then
+    command.component = 'light'
+    dev = device:get_parent_device()
+  end
+  if command.component == 'light' then
+    dev:send(OnOff.server.commands.On(dev))
+  else
+    local speed = dev:get_field('LAST_FAN_SPD') or 1
+    dev:send(FanControl.attributes.FanMode:write(dev,speed))
+    dev:send(FanControl.attributes.FanMode:read(dev,speed))
+  end
+  dev:send(FanControl.attributes.FanMode:read(dev,speed)) 
+end
+
+local function off_handler(driver, device, command)
+  local dev = device
+  if device.network_type == stDevice.NETWORK_TYPE_CHILD then
+    command.component = 'light'
+    dev = device:get_parent_device()
+  end
+  if command.component == 'light' then
+    dev:send(OnOff.server.commands.Off(dev))
+  else
+    dev:send(FanControl.attributes.FanMode:write(dev,FanControl.attributes.FanMode.OFF)) 
+  end
+  dev:send(FanControl.attributes.FanMode:read(dev,FanControl.attributes.FanMode.OFF))
+end
+
+local function switch_level_handler(driver, device, command)
+  local dev = device
+  if device.network_type == stDevice.NETWORK_TYPE_CHILD then
+    command.component = 'light'
+    dev = device:get_parent_device()
+  end
+  if command.component == 'light' then
+    local level = math.floor(command.args.level/100.0 * 254)
+    dev:send(Level.server.commands.MoveToLevelWithOnOff(dev, level, command.args.rate or 0xFFFF))
+  else
+    local speed = level_to_speed(command.args.level)
+    dev:send(FanControl.attributes.FanMode:write(dev,speed))
+    dev:send(FanControl.attributes.FanMode:read(dev,speed))
+  end
+end
+
+local function fan_speed_handler(driver, device, command)
+  device:send(FanControl.attributes.FanMode:write(device,command.args.speed))
+  device:send(FanControl.attributes.FanMode:read(device,command.args.speed))
+end
+
+-- ZIGBEE HANDLERS
+
+local function zb_fan_control_handler(driver, device, value, zb_rx)
+  device:emit_event(capabilities.fanSpeed.fanSpeed(value.value))
+  local evt = capabilities.switch.switch(value.value > 0 and 'on' or 'off', { visibility = { displayed = false } })
+  device:emit_component_event(device.profile.components.main,evt)
+  device:emit_component_event(device.profile.components.main,capabilities.switchLevel.level(levels_for_speed[value.value], { visibility = { displayed = false } }))
+  if value.value > 0 then
+    device:set_field('LAST_FAN_SPD', value.value, {persist = true})
+  end
+end
+
+local function zb_level_handler(driver, device, value, zb_rx)
+  local evt = capabilities.switchLevel.level(math.floor((value.value / 254.0 * 100) + 0.5))
+  device:emit_component_event(device.profile.components.light,evt)
```

Comment:
so this `light` component is not visible to the user as you have this set up, but you're still emitting events here then duplicating them to the child